swarm/src/dummy.rs

```diff
@@ -51,6 +57,23 @@ impl NetworkBehaviour for Behaviour {
             | FromSwarm::ExpiredExternalAddr(_) => {}
         }
     }
+
+    fn on_connection_handler_event(
+        &mut self,
+        _: PeerId,
+        _: ConnectionId,
+        event: THandlerOutEvent<Self>,
+    ) {
+        void::unreachable(event)
+    }
+
+    fn poll(
+        &mut self,
+        _: &mut Context<'_>,
+        _: &mut impl PollParameters,
+    ) -> Poll<NetworkBehaviourAction<Self::OutEvent, THandlerInEvent<Self>>> {
+        Poll::Pending
+    }
```

Comment:
TODO: Check that this diff improves with https://github.com/libp2p/rust-libp2p/pull/3368.